### PR TITLE
list unspent for all addrs and set utxo respectively

### PR DESCRIPTION
**Pull request by nodestory**

### CODE REVIEW

File: command/bitmark-wallet/coins.go [hunk 1]
```diff
@@ -141,8 +141,9 @@ func NewCoinCmd(coinType, short, long string, ct wallet.CoinType) *cobra.Command
 
 			var a agent.CoinAgent
 			switch agentData.Type {
-			case "blockr":
-				a = agent.NewBlockrAgent(agentData.Node)
+			// TODO
```
Comment (nodestory) [inline command/bitmark-wallet/coins.go]: Is it OK not to support blockr?

File: agent/daemon.go [hunk 2]
```diff
@@ -193,29 +161,60 @@ func (da DaemonAgent) GetAddrUnspent(addr string) ([]*tx.UTXO, error) {
...
-		return nil, fmt.Errorf("fail to list all utxos from address: %s", err.Error())
+	// no unspent tx, check if it is an empty address
+	if used, err := da.isAddressUsed(addr); err != nil {
+		return err
+	} else if !used {
+		return ErrNoTxForAddr
 	}
-	if len(utxos) > 0 {
-		return utxos, nil
+	return ErrNoUnspentTx
```
Comment (lemonlatte) [inline agent/daemon.go]:
```suggestion
	return nil
```
The `ErrNoUnspentTx` is useless now.

Comment (lemonlatte) [reply agent/daemon.go]: The error `agent.ErrNoUnspentTx` in the switch case at about LINE 213 could also be removed.

File: agent/daemon_test.go [hunk 3]
```diff
@@ -9,9 +9,8 @@ import (
 func TestDaemonGetAddr(t *testing.T) {
```
Comment (lemonlatte) [inline agent/daemon_test.go]:
```suggestion
func TestDaemonWatchAddress(t *testing.T) {
```

**Pull request merged**